pkg/csi/service/wcp/controller.go

```diff
@@ -495,7 +492,7 @@ func (c *controller) createBlockVolume(ctx context.Context, req *csi.CreateVolum
 				return nil, csifault.CSIInternalFault, logger.LogNewErrorCodef(log, codes.Internal,
 					"failed to find shared datastores for given topology requirement. Error: %v", err)
 			}
-		} else {
+		} else if !hostnameLabelPresent {
```

Comment:
Shouldn't it be `!zoneLabelPresent` here?

Reply:
Is it possible to have that case if TKGs HA is enabled? If yea, then I can change it.

My rationale behind checking for hostNameLabel is so that this logic won't be executed for vDPp flows - where we would have both and we derive required data from the storage pool.

Reply:
This logic was earlier running for single zone supervisor (vdpp & non-vdpp workloads). By making this change, aren't you breaking that workflow?

If you don't need this code to run for vdpp on stretched supervisor, you should just use `!zoneLabelPresent`.

I will leave it unto you to evaluate the impact.
cc: @skogta @divyenpatel

Reply:
Based on the offline discussion with @divyenpatel and @skogta, I've updated the MR. PTAL.